packages/web-components/src/calendar/calendar.ts

```diff
@@ -0,0 +1,368 @@
+   * @param year - year of the month picker
+   * @returns - an object with data about the current and 2 surrounding years
+   * @public
+   */
+  public getMonthPickerInfo(year: number = this.monthPickerYear): MonthPickerInfo {
+    return {
+      year,
+      previous: year - 1,
+      next: year + 1,
+    };
+  }
+
+  /**
+   * Gets data needed to render about a month picker year as well as the previous and next years
+   * @param decadeStart - the start of the decade on the year picker
+   * @returns - an object with data about the current and 2 surrounding decades
+   * @public
+   */
+  public getYearPickerInfo(decadeStart: number = this.yearPickerDecade): YearPickerInfo {
+    return {
+      decadeStart,
+      decadeEnd: decadeStart + 11,
+      previousStart: decadeStart - 12,
+      nextStart: decadeStart + 12,
+    };
+  }
+
+  /**
+   * Changes the month and year on the calendar
+   * @param month - the month to be switched to
+   * @param year - the year to be switched to
+   */
+  public switchMonth(month: number, year: number) {
+    this.selectedDates = '';
+
+    this.year = year;
+    this.month = month;
+
+    this.monthPickerYear = year;
+    this.yearPickerDecade = year - (year % 10);
+  }
+
+  /**
+   * Open/close the year picker
+   * @public
+   */
+  public toggleYearPicker() {
+    this.yearPickerOpen = !this.yearPickerOpen;
+    this.yearPickerDecade = this.monthPickerYear - (this.monthPickerYear % 10);
+  }
+
+  /**
+   * Creates a class string for the "Go to today" link
+   * @returns - string of class names
+   * @public
+   */
+  public getLinkClassNames() {
+    const today: Date = new Date();
+
+    //when the month picker is not visible, this.monthPickerYear is always going to be the current year.
+    return this.month === today.getMonth() + 1 &&
+      this.year === today.getFullYear() &&
+      this.monthPickerYear === today.getFullYear()
+      ? 'slotted-link inactive'
+      : 'slotted-link';
+  }
+
+  /**
+   * Creates a class string for cells on the right panel
+   * @returns - string of class names
+   * @public
+   */
+  public getRightCellClassNames(detail: number, todayMonth: number, todayYear: number) {
+    const isToday = this.yearPickerOpen ? detail === todayYear : detail === todayMonth;
+
+    return ['right-cell-outer', this.hasAttribute('highlightCurrentMonth') && isToday && 'right-panel-today']
+      .filter(Boolean)
+      .join(' ');
+  }
+
+  /**
+   * Returns a list of month labels
+   * @returns A 2D array of month texts
+   * @public
+   */
+  public getMonthText(): { text: string; detail: number }[][] {
+    const months = this.dateFormatter.getMonths();
+    const monthsText: { text: string; detail: number }[][] = [];
+    let monthCount = 0;
+
+    while (monthCount < months.length || monthsText[monthsText.length - 1].length % 4 !== 0) {
+      const month = { text: months[monthCount], detail: monthCount + 1 };
+      const target = monthsText[monthsText.length - 1];
```

Comment:
What does `target` refer to here? We might want to make it more descriptive, e.g. `currentRow` or something similar.

Reply:
Michelle already fixed this